CRM/ManualDirectDebit/Form/SetUp.php

```diff
@@ -70,6 +72,68 @@ public function buildQuickForm() {
 
   public function postProcess() {
     parent::postProcess();
+
+    $values = $this->exportValues();
+
+    $contributionRecur = civicrm_api3('Contribution', 'getsingle', [
```

Comment:
I try to use a rule of thumb to try and keep methods under 50 lines of code, I think this one is getting up close to that amount. Might be a good idea to refactor these API calls into ither methods to improve readibility and separate concerns. This one in particular got me confused, as the variable's name is `$contributionRecur`, but the call is to 'Contribution' entity initially. So I though, why did he call the variable that way? But then I realized there was a linked API call to obtain the recurring contribution.

Eg:
```php
public function postProcess() {
  parent::postProcess();

  $values = $this->exportValues();

  $contributionRecur = $this->getRecurringContribution($values['contribution_id']);
  $this->updateRecurringContribution($contributionRecur, $values);
  $this->createDirectDebitMandate($contributionRecur, $values);

  $url = CRM_Utils_System::url('civicrm/direct_debit/setup/confirmation');
  CRM_Utils_System::redirect( $url);
}
```

Reply:
@MiyaNoctem the reason I am doing that because I don't want APIs twice.